Assets/Source/Components/Enemy/Base/EnemyAIBase.cs

```diff
@@ -0,0 +1,78 @@
+﻿using UnityEngine;
+using Assets.Source.Components.Base;
+using Assets.Source.Constants;
+using Assets.Source.Components.Actor;
+using Assets.Source.Extensions;
+
+namespace Assets.Source.Components.Enemy.Base
+{
+    public abstract class EnemyAIBase : ComponentBase
+    {
+        private readonly float STABILIZATION_RATE = 0.01f;
+
+        private GameObject explosionPrefab;
+
+        protected Vector2 externalVelocity;
+        protected Rigidbody2D rigidBody;
+        protected ActorBehavior actorBehavior;
+        protected GameObject player;
+
+        public override void ComponentAwake()
+        {
+            //todo: design a better way to have a reference to the play since every spawned enemy will have to try and "find" the player
```

Comment:
yeah i agree.  I have yet to think of a better way though.  If you've got any ideas pls share

Reply:
The only other decent way i know of is to have a global reference to it so like putting a reference of the player instance in a class that we can guarantee gets created right away in every scene. Or something like that